apps/emqx/src/emqx_cm_registry.erl

```diff
@@ -91,18 +97,45 @@ unregister_channel(ClientId) when is_binary(ClientId) ->
     unregister_channel({ClientId, self()});
 unregister_channel({ClientId, ChanPid}) when is_binary(ClientId), is_pid(ChanPid) ->
     case is_enabled() of
-        true -> mria:dirty_delete_object(?CHAN_REG_TAB, record(ClientId, ChanPid));
-        false -> ok
+        true ->
+            mria:dirty_delete_object(?CHAN_REG_TAB, record(ClientId, ChanPid)),
```

Comment:
Maybe unite deletion & history insertion in a single dirty activity?

Reply:
how to achieve that ?